spec/models/ability_spec.rb

```diff
@@ -66,152 +66,142 @@
     SolrDocument.new(solr_rsp['response']['docs'].first)
   end
 
-  after do
+  after(:all) do
     cleanup_jetty
   end
 
   describe "a user with no roles" do
-    let(:guest) { nil }
-    subject { Ability.new(guest) }
-
-    it { is_expected.not_to be_able_to(:create, GenericFile) }
-    it { is_expected.not_to be_able_to(:edit, file) }
-    it { is_expected.not_to be_able_to(:update, file) }
-    it { is_expected.not_to be_able_to(:destroy, file) }
-
-    it { is_expected.not_to be_able_to(:create, Collection) }
-    it { is_expected.not_to be_able_to(:edit, collection) }
-    it { is_expected.not_to be_able_to(:update, collection) }
-    it { is_expected.not_to be_able_to(:destroy, collection) }
-
-    it { is_expected.not_to be_able_to(:create, TinymceAsset) }
-    it { is_expected.not_to be_able_to(:update, ContentBlock) }
-
-    it {is_expected.not_to be_able_to(:download, restricted_file) }
-    it {is_expected.not_to be_able_to(:download, restricted_doc) }
-    it { is_expected.to be_able_to :read, institutionally_restricted_file}
-    it {is_expected.not_to be_able_to(:download, institutionally_restricted_file) }
-    it {is_expected.not_to be_able_to(:download, institutionally_restricted_doc) }
+    let(:guest_abilities) { Ability.new(nil) }
+
+    it 'has proper permissions' do
+      expect(guest_abilities).not_to be_able_to(:create, GenericFile)
+      expect(guest_abilities).not_to be_able_to(:edit, file)
+      expect(guest_abilities).not_to be_able_to(:update, file)
+      expect(guest_abilities).not_to be_able_to(:destroy, file)
+
+      expect(guest_abilities).not_to be_able_to(:create, Collection)
+      expect(guest_abilities).not_to be_able_to(:edit, collection)
+      expect(guest_abilities).not_to be_able_to(:update, collection)
+      expect(guest_abilities).not_to be_able_to(:destroy, collection)
+
+      expect(guest_abilities).not_to be_able_to(:create, TinymceAsset)
+      expect(guest_abilities).not_to be_able_to(:update, ContentBlock)
+
+      expect(guest_abilities).not_to be_able_to(:download, restricted_file)
+      expect(guest_abilities).not_to be_able_to(:download, restricted_doc)
+      expect(guest_abilities).to be_able_to(:read, institutionally_restricted_file)
+      expect(guest_abilities).not_to be_able_to(:download, institutionally_restricted_file)
+      expect(guest_abilities).not_to be_able_to(:download, institutionally_restricted_doc)
+    end
   end
 
   describe "a registered user" do
-    subject { Ability.new(user) }
```

Comment:
I disagree with subject.

- implicit subject should never be used. (where rspec makes assumptions on the subject and bases it off an instantiation of the described class): Even the rspec docs will say this: https://www.relishapp.com/rspec/rspec-core/v/3-5/docs/subject/implicitly-defined-subject
- Explicit subject is better but you should be naming the subject. Rspec docs will tell you to do this as well: https://www.relishapp.com/rspec/rspec-core/v/3-5/docs/subject/explicit-subject.

 But in this case I would argue....why not just use a let block instead? Which is what I did here